event_routing_backends/processors/mixins/base_transformer.py

```diff
@@ -102,3 +102,16 @@ def transform(self):
                 )
 
         return self.transformed_event
+
+    def extract_username(self):
+        """
+        extract username edX event.
+
+        Returns:
+            username
+        """
+        username = self.event['context']['username']
```

Comment:
can we simplify these statement to something like this?
`return self.event['context'].get('username', self.event['data'].get('username'))`

Reply:
'str' object has no attribute 'get'

Reply:
self.event['context'].get('username', self.event['data'].get('username'))
This line of code will return username in data dict if username key is not set in context dict. But in our case username key is set in the context dict but it's value is empty string ''. so we need to have a check if user name in context dic is empty as well as it exist in data dict. So I have added above code that is working fine